program-runtime/src/loaded_programs.rs

```diff
@@ -2386,6 +2375,38 @@ mod tests {
         assert!(match_missing(&missing, &program3, true));
     }
 
+    #[test]
+    fn test_unloaded() {
```

Comment:
This looks good, but we still don't have tests for the None path in
sort_and_unload and evict_using_2s_random_selection?